micro-frontends-with-angular-and-stenciljs.md

```diff
@@ -0,0 +1,150 @@
+
+## Finding the right WebComponent framework
+
+After that decision, we needed to find a suitable framework to build the WebComponents. It is possible to build WebComponents with Angular by using Angular Elements. Using Angular would be a good fit because the whole team knows and is developing Angular. Unfortunately, Angular Elements build a complete Angular application into the WebComponents, which results in rather large bundles.
+
+I compared the bundle sizes a while ago. I used [ngx-build-plus](https://github.com/manfredsteyer/ngx-build-plus) to build a single file Angular Elements bundle. Stencil.js outputs a bundle per WebComponent. 
+
+For my example, I got the following values (round in kB):
+
+* Angular: 186 kB
+* Stencil.js: 13kB
+
+This is quite shocking. The Stencil.js bundle size is only around 24% of Angular Elements bundle size.
+
+With this result, we decided to implement our layout components with Stencil.js.
+
+## Create and Stencil project
+
+Let's have a look at our setup with StencilJS and the integration in our Angular applications. We are using [NX](https://nx.dev/) for our UI workspaces. Let's create an NX workspace with a StencilJS project.
+
+First, we create the NX workspace with typescript presets:
+
+```bash
+npx create-nx-workspace my-workspace --preset=ts
+```
+
+Afterward we are going to create a Stencil project in our freshly created NX workspace by using the NX plugin from [Nnext](https://nxext.dev/docs/nxext/overview.html):
+
+```bash
+npm install @nxext/stencil --save-dev
+nx g @nxext/stencil:library my-lib
+```
+
+The Nnext plugin provides a generator that allows to create our first component:
+
+```bash
+nx g @nxext/stencil:component my-header --project my-lib
+```
+
+This outputs a component skeleton like this:
+
+```ts
+@Component({
+  tag: 'my-header',
+  styleUrl: 'my-header.scss',
+  shadow: true,
+})
+export class MyHeader {
+  @Prop() first: string;
+  @Prop() middle: string;
+  @Prop() last: string;
+
+  private getText(): string {
+    return (this.first || '') + (this.middle ? ` ${this.middle}` : '') + (this.last ? ` ${this.last}` : '');
+  }
+
+  render() {
+    return <div>Hello, World! I'm {this.getText()}</div>;
+  }
+}
+```
+
+I will not be going to write a StencilJS tutorial at this place. But one thing I need to mention. Stencil provides several [output targets](https://stenciljs.com/docs/output-targets). We actually want to have a single file bundle which would be the `dist-custom-elements-bundle` out target. Unfortunately, this has been deprecated. We use its successor `dist-custom-elements`. It does basically the same thing but produces a js file per component. This is the better way because it is better to load several smaller bundles than a large one. It will also be better for leveraging `http2`. We would prefer a single file because it means less file handling in our frontend's, but we can work with that.
```

Comment:
More of a general comment: To make reviewing and adding comments easier, we should wrap long lines. Prettier can do this automatically with the { proseWrap: "always" } setting. We already have an `.editorconfig` file, but I think we should add a `.prettierrc` file with that setting. editorconfig requires a separate extension. The Prettier extension pretty much everyone already has installed.

Reply:
A good suggestion, can you introduce a .prettierrc? I'll not be going to do it with that PR, we should not mix things in a PR.